### ISSUE TITLE
Support for multiple subnets for static_ips

### ISSUE THREAD

**Issue by mtekel** [geofffranks/spruce#124]
Hi,

with new bosh 2.0, you can specify all availability zone subnets directly in your network and then control which zone the instance gets deployed to directly in job specification. If you have static IP specified for the given job (or instance_group), then BOSH would figure out which zones/subnets those IPs belong to and deploy the VMs there.

Spruce currently picks up static ip information just from the 1st subnet of the multi-subnet network definition. This means that if you use static_ips, it will generate IPs only from that subnet forcing BOSH to deploy to single zone, which is not what you always want.

The solution is to either add some parameter to static_ips to tell it which subnet to use for calculation, or some feature to make it spread static IPs among all subnets...

see http://play.spruce.cf/#1b850fa29430383c1175c681953123a4 for example manifest. I'd like to use all zones for staticIP job.

**Comment by geofffranks** [comment]
Currently spruce does look at all the addresses. Debug output from your example shows a static IP pool from both 192.168.1.0/24 and 192.168.0.0/24. Since spruce has no concept of AZs though, it doesn't know that it should split the IPs across the networks for a job, rather than just take one per subnet.

```
 DEBUG>   checking to see if the caller asked for enough static_ips to provision
           → all job instances (need at least 2)
 DEBUG>   looks good.  asking for 3 IPs for a job with 2 instances
 
 DEBUG>   determining the pool of static IPs from which to provision
 DEBUG>   static IP pool: [192.168.1.2 192.168.1.3 192.168.1.4 192.168.1.5
           → 192.168.1.6 192.168.1.7 192.168.1.8 192.168.1.9 192.168.1.10 192.168.0.2
           → 192.168.0.3 192.168.0.4 192.168.0.5 192.168.0.6 192.168.0.7 192.168.0.8
           → 192.168.0.9 192.168.0.10]
 DEBUG>   found 18 addresses in the pool
```

Should spruce be changed to interleave the IPs from each subnet, for a round-robin allocation, rather than preferring to exhaust the first subnet and moving on? Do we need to reverse-engineer how BOSH divvies up instances across AZs when static IPs are specified, and implement something specific to that?

**Comment by mtekel** [comment]
It would be nice if there was option to round robin it. No need to reverse engineer BOSH as you control where it deploys by being explicit with IPs.

**Comment by geofffranks** [comment]
Is there any disadvantage to *always* round-robin-ing the subnets?

**Comment by mtekel** [comment]
I don't mind, but it's a change in behaviour and might not be backward compatible? Someone could have a manifest that depends on the old behaviour? It could be rare though. You could do it and simply just announce the change...

**Comment by jhunt** [comment]
One idea was to extend the indexing syntax of the `(( static_ips ... ))` operator so that you could be explicit about AZ usage:

```
(( static_ips az1:0 az2:0 az1:1 az2:1 ))
```

This may also dovetail nicely with some other work we've thought about doing, allowing ranges of IPs:

```
(( static_ips 0-4 10-19 ))
```

which extends quite nicely into az-territory:

```
(( static_ips az1:0-4 az2:0-3 ))
```

**Comment by geofffranks** [comment]
I like this approach.

**Comment by JamesClonk** [comment]
Would love to have that feature too! 👍

**Comment by geofffranks** [comment]
Fixed as of v1.7.0
